grudge/geometry/metrics.py

```diff
@@ -408,20 +406,24 @@ def inverse_surface_metric_derivative(
     if dd is None:
         dd = DD_VOLUME
     dd = dof_desc.as_dofdesc(dd)
+    dd_base = dd.with_discr_tag(DISCR_TAG_BASE)
 
     if ambient_dim == dim:
         result = inverse_metric_derivative(
-            actx, dcoll, rst_axis, xyz_axis, dd=dd
+            actx, dcoll, rst_axis, xyz_axis, dd=dd_base
         )
     else:
-        inv_form1 = inverse_first_fundamental_form(actx, dcoll, dd=dd)
+        inv_form1 = inverse_first_fundamental_form(actx, dcoll, dd=dd_base)
         result = sum(
             inv_form1[rst_axis, d]*forward_metric_nth_derivative(
-                actx, dcoll, xyz_axis, d, dd=dd
+                actx, dcoll, xyz_axis, d, dd=dd_base
             ) for d in range(dim))
```

Comment:
I think these are nonlinear expressions and thus I think it's better to evaluate them on the quadrature grid. If you agree, maybe add a comment to that effect?

Reply:
> I think it's better to evaluate them on the quadrature grid. If you agree, maybe add a comment to that effect?

I mean I would agree in principle if it was even possible to do this in `meshmode`. The metric derivatives (even before this PR) are always computed on the nodal grid since `num_reference_derivative` can only be evaluated on the interpolation grid.

In short, `meshmode` reference derivative computations are only ever computed on the nodal grid, see: https://github.com/inducer/meshmode/blob/main/meshmode/discretization/__init__.py#L614-L618. Hence why `dd_base` is always used to compute the metric derivatives.

So this comment doesn't really make much sense given our current infrastructure. All I have done in this PR is move where the interpolation happens.